env_spec.py

```diff
@@ -19,138 +28,216 @@
+    return choice_str
+
 
-    if "[" in input_type and "=" in input_type:
-        env_var_type, value = input_type.split("=")
-        value = value.strip()
+def render_to_html(env_spec_list):
+    """
+    Creates the html output from the env_spec list, by checking whether the
+    value is None or not.(comments, choices, default_value)
+    Returns the html output of the whole list.
+    """
+    if not env_spec_list:
+        return []
+    html_output = ""
 
-        if value in env_var_type:
-            return True
+    for env_spec_entry in env_spec_list:
+        if env_spec_entry["choices"] is None:
+            ret_str = render_label(env_spec_entry)
+            ret_str += render_input(env_spec_entry)
         else:
-            return False
-    if "=" in input_type:
-        return True
-
-    if "[" in input_type:
-        return True
-    elif input_type not in valid_types_list:
-        return False
-    else:
-        return True
+            ret_str = render_label(env_spec_entry)
+            ret_str += f'<select id="env_spec_{env_spec_entry["name"].lower()}" name="{env_spec_entry["name"].lower()}">\n'
 
+            for choice in env_spec_entry["choices"]:
+                ret_str += render_choice(
+                    choice, choice == env_spec_entry["default_value"]
+                )
+            ret_str += "</select>\n"
 
-def render_env_var_spec(env_var_field, env_var_type):
+        if env_spec_entry["comment"] is not None:
+            ret_str += f"<small>{env_spec_entry['comment']}</small>\n"
+
+        html_output += ret_str
+    return html_output
+
+
+def parse(env_spec_text):
     """
-    Creates html string from env_variables (field, type). If env_var_type is a restricted choice,
-    calls create_list to create the list of strs from env_var_type string. Returns the string.
+    Parse function creates a list of dictionaries by splitting env_spec_text by
+    \n. Every line will be a dictionary.The keys of dictionary are:
+    name, type, choices, default_value, comment.
+    Returns the env_spec listm or empty list [], for exception.
     """
+    env_spec_list = []
 
-    env_var_field_lower = env_var_field.lower()
-    env_var_type = env_var_type or "text"
+    alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
+    name_regex = r"^(.+)\:(.+)$"
+    choices_regex = r"^(.+)\]"
+    default_values_regex = r"^(.+)\=(.+)$"
+    line_comment_regex = r"^#"
+    comment_regex = r"^(.+)\#(.+)$"
 
-    if "[" not in env_var_type:
-        if "=" in env_var_type:
-            env_var_type, value = env_var_type.split("=")
-            value = value.strip()
+    lines = env_spec_text.split("\n")
 
-            ret_str = (
-                f'<label for="env_spec_{env_var_field_lower}">{env_var_field}</label>\n'
-                f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" type="{env_var_type}" value="{value}" />\n'
-            )
+    for line in lines:
+        line_dict = {}
```

Comment:
Instead of using a generic name like `line_dict` why not use something more semantic like `env_spec_entry` ✨?